python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -717,7 +721,11 @@ def get_2d_run_length_encoded_partitions(
                 if isinstance(primary_partitions_def, TimeWindowPartitionsDefinition):
                     time_windows = cast(
                         TimeWindowPartitionsDefinition, primary_partitions_def
-                    ).time_windows_for_partition_keys(frozenset([start_key, end_key]))
+                    ).time_windows_for_partition_keys(
+                        frozenset([start_key, end_key]),
+                        current_time=current_time,
+                        validate=False,  # we already know these keys are in the partition set
```

Comment:
this actually fixes it in two different ways - just sending in validate=False would be sufficient since the slow uncached get_last_partition_window call was in the validation code. But even if it was True, making sure it stayed cached would also help.